app/views/add-reduction.html

```diff
@@ -0,0 +1,156 @@
+          </div>
+          <div class="form-group form-group-year" id="reduction_start_year">
+            <label class="form-label" for="start-year">Year</label>
+            <input class="form-control" id="start-year" name="red_start_year" type="number" min="0" max="2099" onchange="calculateExpiry()">
+          </div>
+        </div>
+      </fieldset>
+    </div>
+
+    <div class="form-group" id="reduction_end_date">
+      <fieldset>
+        <legend>
+          <span class="form-label-bold">Reduction end date</span>
+          <span class="form-hint">For example, 31 3 1980</span>
+        </legend>
+        <div class="form-date" id="reduction_end_day">
+          <div class="form-group form-group-day">
+            <label class="form-label" for="end-day">Day</label>
+            <input class="form-control" id="end-day" name="red_end_day" type="number" pattern="[+-]?[0-9]{1,2}" min="1" max="31">
+          </div>
+          <div class="form-group form-group-month" id="reduction_end_month">
+            <label class="form-label" for="end-month">Month</label>
+            <input class="form-control" id="end-month" name="red_end_month" type="number" pattern="[+-]?[0-9]{1,2}" min="1" max="12">
+          </div>
+          <div class="form-group form-group-year" id="reduction_end_year">
+            <label class="form-label" for="end-year">Year</label>
+            <input class="form-control" id="end-year" name="red_end_year" type="number" pattern="[+-]?[0-9]{1,4}" min="0" max="">
+          </div>
+        </div>
+      </fieldset>
+    </div>
+
+    <div class="form-group">
+      <label class="form-label-bold" for="textarea">Additional notes</label>
+      <textarea class="form-control form-control-3-4" name="notes" id="textarea" rows="5"></textarea>
+    </div>
+    <input type="hidden" name="reasonForReductionId" id="hiddenId" value="0"> 
+    {% include "includes/csrf-hidden-input.html"%} 
+    <input class="button" type="submit" value="Add reduction">
+  </form>
+
+</main>
+{% endblock %} {% block body_end %}
+<script type="text/javascript">
+  var refData = {{ referenceData | dump | safe }}
+
+  var chosenReduction
+
+  var populateHours = function () {    
+    var value = ''
+    var hiddenIdValue
+
+    var chosenReductionIndex = document.getElementById("select-box").selectedIndex
+
+    chosenReduction = refData[chosenReductionIndex]
+    
+    if (chosenReduction.allowancePercentage !== null) {
+      document.getElementById('hours').value = chosenReduction.allowanceHours
+      document.getElementById('hours').setAttribute('readonly', true)
+    } else {
+      document.getElementById('hours').value = value
+      document.getElementById('hours').removeAttribute('readonly')
+    }
+    
+    if (chosenReduction.maxAllowancePercentage !== null) {
+      document.getElementById('hours').setAttribute('max', chosenReduction.maxAllowanceHours)
+    }
```

Comment:
Need to unset this max if another reason with a free text box is selected after one that had a max

Reply:
👍

Reply:
Was this added?

Reply:
Good spot again.